src/SSVOpenHexagon/Components/CWall.cpp

```diff
@@ -62,15 +65,48 @@ void CWall::update(
         }
         else
         {
+            if(moveTowardsCenter(mHexagonGame, mFT, vp, mCenterPos)){
+                player.kill(mHexagonGame, true);
+            }
+        }
+        if(moveCurve(mHexagonGame, mFT, vp, mCenterPos) && curve.speed != 0.f){
+            player.push(mHexagonGame, *this);
+        }
+    }
+
+    if(pointsOnCenter > 3)
+    {
+        killed = true;
+    }
+    */
+}
+
+bool CWall::moveTowardsCenter(HexagonGame& mHexagonGame, const sf::Vector2f& mCenterPos, ssvu::FT mFT){
+    const float radius{mHexagonGame.getRadius() * 0.65f};
+    int pointsOnCenter{0};
+    for(sf::Vector2f& vp : vertexPositions) {
+        if (std::abs(vp.x - mCenterPos.x) < radius &&
+            std::abs(vp.y - mCenterPos.y) < radius) {
+            ++pointsOnCenter;
+        } else {
             ssvs::moveTowards(vp, mCenterPos, speed.speed * 5.f * mFT);
-            ssvs::rotateRadAround(vp, mCenterPos, curve.speed / 60.f * mFT);
         }
     }
 
     if(pointsOnCenter > 3)
     {
         killed = true;
     }
+
+    return isOverlapping(mHexagonGame.player.getPosition());
+}
+
+bool CWall::moveCurve(HexagonGame& mHexagonGame, const sf::Vector2f& mCenterPos, ssvu::FT mFT){
```

Comment:
Ditto, should return `void` and we should check collision directly in the loop in `HexagonGame::updateWalls`